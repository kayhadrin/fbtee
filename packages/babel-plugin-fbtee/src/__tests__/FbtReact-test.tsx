import { describe, expect } from '@jest/globals';
import { jsCodeFbtCallSerializer, snapshotTransform } from './FbtTestUtil.tsx';

expect.addSnapshotSerializer(jsCodeFbtCallSerializer);

describe('Fbt React Component transform:', () => {
  test('supports simple <Fbt> string', () => {
    expect(
      snapshotTransform(
        `import { Fbt } from "Fbtee.react";
        <Fbt desc="some-desc">
          A simple string
        </Fbt>`,
      ),
    ).toMatchSnapshot();
  });
});
