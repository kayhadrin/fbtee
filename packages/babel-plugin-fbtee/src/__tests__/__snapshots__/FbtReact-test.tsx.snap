// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Fbt React Component transform: supports simple <Fbt> string 1`] = `
import { fbt } from "fbtee";
import { FbtImpl } from "Fbtee.react";
/*#__PURE__*/ React.createElement(
  FbtImpl,
  {
    desc: "some-desc",
  },
  "A simple string"
);

`;
